### ISSUE TITLE
Printing with Firefox does not fill all the page with the slide

### ISSUE THREAD

**Issue by a-i-m-a-r** [hakimel/reveal.js#3296]
I now that Chrome and Chromium are recommended for printing purposes, but maybe because my presentation is quite heavy (>150 pages) the preview rendering took ages and usually crashes.
So I am trying to use Firefox instead.
I have some issue regarding the slide numbers which is mentioned in [another topic](https://github.com/hakimel/reveal.js/issues/3295) (#3295).
However I am facing another issue. Firefox do not use all the paper to print the slide. It is like it scales the slide even though I selected "Fit to page width". 
![image](https://user-images.githubusercontent.com/15641960/194512066-32b38174-7c7c-436e-80f9-78a33c4e4b11.png)

How could I tweak the printing size? I now it is quite dirty what I am trying to achieve but, although I am not a developer, I also enjoy digging and rummaging the code and customize some bits of it.

**Comment by Benson665** [comment]
Am impressed with your work good to floor with

**Comment by a-i-m-a-r** [comment]
I though that modifying reveal.js/js/controllers/print.js would do the trick. But it does not.
The following lines should in my inexpert opinion control the page size and the slide size within the page, but even though I modify manually them, nothing happens.

```
// Dimensions of the PDF pages
let pageWidth = Math.floor( slideSize.width * ( 1 + config.margin ) ),
pageHeight = Math.floor( slideSize.height * ( 1 + config.margin ) );

// Dimensions of slides within the pages
let slideWidth = slideSize.width,
slideHeight = slideSize.height;
```
Although I manage to resolve the page number issue in Chromium, Chromium is not a good option for me. My presentation contains two columns with unordered lists that Firefox split correctly in two columns while Chromium does not, displacing the rest of the content out of the slide, so the printed document lacks part of the information.

**Comment by a-i-m-a-r** [comment]
I found that I could set the following configuration options to fill all the page, but the content is not scaled, just the background, the toc-progress, etc. so it does not look properly.
```
Reveal.initialize({
  width: 1169,
  height: 827,
});
```

**Comment by manzo2202** [comment]
ในวันที่ จ. 14 พ.ย. 2022 17:04 น. a-i-m-a-r ***@***.***>
เขียนว่า:

**Comment by manzo2202** [comment]
ในวันที่ จ. 14 พ.ย. 2022 21:32 น. Arm L ***@***.***> เขียนว่า:

**Comment by a-i-m-a-r** [comment]
Finally, I did print the presentation with those big white margins. Then I cropped it: `pdfcrop --margins "0 0 0 0" margins.pdf cropped.pdf`. Afterwards I rescaled the document to fit a DIN A4: `gs -o a4.pdf -sDEVICE=pdfwrite -r300 -g3508x2480 -dPDFFitPage cropped.pdf.` And with latex, I inserted the page numbers. Now it looks nice. However, I think I will modify height and width as mentioned above to fit the DIN A4 size. I will have to rearrange the layout but in the screen the presentation will be scaled to fit the projector resolution, so no problem, and I will save the cropping and resizing steps for printing. I just will add the page numbers.
